### PR TITLE
Clarify the documentation of GridGenerator::cylinder().

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

**Comment by tjhei** [comment]
I am not sure this new explanation is a lot more clear than before. I mean, this function would most likely be used only in 3d so having this as the main definition in the first paragraph makes more sense to me.

File: include/deal.II/grid/grid_generator.h [hunk 1]
```diff
@@ -713,13 +713,16 @@ namespace GridGenerator
...
-   * In two dimensions, the cylinder is a rectangle from
-   * <tt>x=-half_length</tt> to <tt>x=+half_length</tt> and from
-   * <tt>y=-radius</tt> to <tt>y=radius</tt>.
+   * Create a @p dim dimensional cylinder where the $x$-axis serves as
+   * the axis of the cylinder. For the purposes of this function, a
+   * cylinder is defined as a (@p dim - 1) dimensional disk of given
+   * @p radius, extruded along the axis of the cylinder (which is the
+   * first coordinate direction). Consequently, in two dimensions, the
+   * cylinder is a rectangle from `x=-half_length` to
+   * `x=+half_length<` and from `y=-radius` to `y=radius`.  In three
```
Comment (drwells) [inline include/deal.II/grid/grid_generator.h]: There is an extra `<`

**Comment by bangerth** [comment]
I fixed the mistake @drwells found, and I've exchanged the 2d and 3d discussions since the idea of the function is indeed to create 3d meshes. @tjhei -- does that address your criticism?

@davydden -- I didn't know where you saw the additional `.`. Can you take another look?

File: include/deal.II/grid/grid_generator.h [hunk 2]
```diff
@@ -713,13 +713,16 @@ namespace GridGenerator
...
-   * <tt>x=-half_length</tt> to <tt>x=+half_length</tt> and its projection
-   * into the @p yz-plane is a circle of radius @p radius.
-   *
-   * In two dimensions, the cylinder is a rectangle from
-   * <tt>x=-half_length</tt> to <tt>x=+half_length</tt> and from
-   * <tt>y=-radius</tt> to <tt>y=radius</tt>.
+   * Create a @p dim dimensional cylinder where the $x$-axis serves as
+   * the axis of the cylinder. For the purposes of this function, a
+   * cylinder is defined as a (@p dim - 1) dimensional disk of given
+   * @p radius, extruded along the axis of the cylinder (which is the
```
[the remark this answers is not in the document]
Comment (davydden) [reply include/deal.II/grid/grid_generator.h]: > Hm, where?

hm, perhaps it was dirt on my screen 😄

**Comment by drwells** [comment]
Since its just a documentation change lets merge :)

**Pull request merged**